src/clarity-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -114,6 +114,33 @@ export default function(): void {
             expect(selectionInstance.isSelected(4)).toBe(false);
         });
 
+        it("handles clrDgItems trackBy index expressions in multi selection type", function() {
+            // IMPORTANT, using clrDgItems with trackBy will use trackBy reference in multi
+            selectionInstance.selectionType = SelectionType.Multi;
+            itemsInstance.trackBy = (index, item) => index;
+            // Must set the trackBy before setting the items
+            itemsInstance.all = ["a", "b", "c", "d", "e"];
+            selectionInstance.current = [1, 3];
```

Comment:
Yeah, that's what I feared. By changing the meaning of `current`, you break the behavior of our two-way binding on our current selection. People want to select the items, not their trackBys.

Reply:
I thought this is what we discussed though. If you have a trackBy, the current becomes an array of the trackBy references (server side), otherwise its the exact items (client side). Looks like the rest of the comments hinge upon this point. If we need to use a different property to maintain the list references so we can track the selection and the raw items, but we'd have duplicates of the arrays for client side data grids.